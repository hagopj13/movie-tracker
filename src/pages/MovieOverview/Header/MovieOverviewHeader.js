// @flow
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Card from '@material-ui/core/Card';

import MovieBackdrop from 'components/Movie/Backdrop/MovieBackdrop';
import MoviePoster from 'components/Movie/Poster/MoviePoster';
import MovieSummary from 'components/Movie/Summary/MovieSummary';
import MovieActions from 'components/Movie/Actions/MovieActions';
import type { Movie } from 'types';

type Props = {
  movie: Movie,
};

const useStyles = makeStyles((theme) => ({
  root: {
    color: theme.palette.common.white,
    padding: theme.spacing(5),
    display: 'flex',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
      alignItems: 'center',
    },
  },
  poster: {
    borderRadius: 8,
    width: 300,
    minWidth: 300,
    height: 450,
    marginRight: theme.spacing(5),
    [theme.breakpoints.down('xs')]: {
      marginRight: 0,
      marginBottom: theme.spacing(2),
    },
  },
  infoContainer: {
    paddingTop: theme.spacing(1),
  },
  actionsContainer: {
    marginTop: theme.spacing(2),
    [theme.breakpoints.down('xs')]: {
      textAlign: 'center',
    },
  },
}));

const MovieOverviewHeader = (props: Props) => {
  const { movie } = props;

  const classes = useStyles();

  return (
    <MovieBackdrop backdropPath={movie.backdropPath}>
      <Container className={classes.root}>
        <Card className={classes.poster} elevation={5}>
          <MoviePoster posterPath={movie.posterPath} height={450} width={300} />
        </Card>
        <div className={classes.infoContainer}>
          <MovieSummary movie={movie} />
          <div className={classes.actionsContainer}>
            <MovieActions movie={movie} />
          </div>
        </div>
      </Container>
    </MovieBackdrop>
  );
};

export default MovieOverviewHeader;
