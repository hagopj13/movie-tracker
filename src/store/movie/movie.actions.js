// @flow
import type { Movie, MovieUserState } from 'types';

import MovieActionTypes from './movie.types';

const fetchMovie = (id: string) => ({
  type: MovieActionTypes.FETCH_MOVIE,
  payload: { id },
});

const fetchMovieStart = () => ({
  type: MovieActionTypes.FETCH_MOVIE_START,
});

const fetchMovieSuccess = (movie: Movie, movieUserState: MovieUserState) => ({
  type: MovieActionTypes.FETCH_MOVIE_SUCCESS,
  payload: { movie, movieUserState },
});

const fetchMovieFailure = (error: string) => ({
  type: MovieActionTypes.FETCH_MOVIE_FAILURE,
  payload: { error },
});

const fetchMovieUserState = (id: string) => ({
  type: MovieActionTypes.FETCH_MOVIE_USER_STATE,
  payload: { id },
});

const fetchMovieUserStateStart = () => ({
  type: MovieActionTypes.FETCH_MOVIE_USER_STATE_START,
});

const fetchMovieUserStateSuccess = (movieUserState: MovieUserState) => ({
  type: MovieActionTypes.FETCH_MOVIE_USER_STATE_SUCCESS,
  payload: { movieUserState },
});

const fetchMovieUserStateFailure = (error: string) => ({
  type: MovieActionTypes.FETCH_MOVIE_USER_STATE_FAILURE,
  payload: { error },
});

const setIsMovieFavorite = (id: number, isFavorite: boolean) => ({
  type: MovieActionTypes.SET_IS_MOVIE_FAVORITE,
  payload: { id, isFavorite },
});

const setIsMovieFavoriteStart = () => ({
  type: MovieActionTypes.SET_IS_MOVIE_FAVORITE_START,
});

const setIsMovieFavoriteSuccess = (id: number, isFavorite: boolean) => ({
  type: MovieActionTypes.SET_IS_MOVIE_FAVORITE_SUCCESS,
  payload: { id, isFavorite },
});

const setIsMovieFavoriteFailure = (error: string) => ({
  type: MovieActionTypes.SET_IS_MOVIE_FAVORITE_FAILURE,
  payload: { error },
});

const setIsMovieInWatchlist = (id: string, isInWatchlist: boolean) => ({
  type: MovieActionTypes.SET_IS_MOVIE_IN_WATCHLIST,
  payload: { id, isInWatchlist },
});

const setIsMovieInWatchlistStart = () => ({
  type: MovieActionTypes.SET_IS_MOVIE_IN_WATCHLIST_START,
});

const setIsMovieInWatchlistSuccess = (id: number, isInWatchlist: boolean) => ({
  type: MovieActionTypes.SET_IS_MOVIE_IN_WATCHLIST_SUCCESS,
  payload: { id, isInWatchlist },
});

const setIsMovieInWatchlistFailure = (error: string) => ({
  type: MovieActionTypes.SET_IS_MOVIE_IN_WATCHLIST_FAILURE,
  payload: { error },
});

const rateMovie = (id: number, rating: number | null) => ({
  type: MovieActionTypes.RATE_MOVIE,
  payload: { id, rating },
});

const rateMovieStart = () => ({
  type: MovieActionTypes.RATE_MOVIE_START,
});

const rateMovieSuccess = (id: number, rating: number | null) => ({
  type: MovieActionTypes.RATE_MOVIE_SUCCESS,
  payload: { id, rating },
});

const rateMovieFailure = (error: string) => ({
  type: MovieActionTypes.RATE_MOVIE_FAILURE,
  payload: { error },
});

export default {
  fetchMovie,
  fetchMovieStart,
  fetchMovieSuccess,
  fetchMovieFailure,
  fetchMovieUserState,
  fetchMovieUserStateStart,
  fetchMovieUserStateSuccess,
  fetchMovieUserStateFailure,
  setIsMovieFavorite,
  setIsMovieFavoriteStart,
  setIsMovieFavoriteSuccess,
  setIsMovieFavoriteFailure,
  setIsMovieInWatchlist,
  setIsMovieInWatchlistStart,
  setIsMovieInWatchlistSuccess,
  setIsMovieInWatchlistFailure,
  rateMovie,
  rateMovieStart,
  rateMovieSuccess,
  rateMovieFailure,
};
