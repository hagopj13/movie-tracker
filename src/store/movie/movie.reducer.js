// @flow
import { handleActions } from 'redux-actions';

import AuthActionTypes from 'store/auth/auth.types';
import type { Movie, MovieUserState } from 'types';

import MovieActionTypes from './movie.types';

type State = {
  details: Movie,
  userState: MovieUserState,
};

export const initialState: State = {
  details: {
    id: null,
    title: '',
    releaseDate: '',
    voteAverage: 0,
    voteCount: 0,
    posterPath: '',
    backdropPath: '',
    tagline: '',
    overview: '',
    runtime: 0,
    status: '',
    genres: [],
    actors: [],
    reviews: [],
    recommendations: [],
  },
  userState: {
    isFavorite: false,
    isInWatchlist: false,
    rating: null,
  },
};

const fetchMovieStart = () => initialState;

const fetchMovieSuccess = (state: State, action) => ({
  ...state,
  details: action.payload.movie,
  userState: action.payload.movieUserState,
});

const fetchMovieUserStateSuccess = (state: State, action) => ({
  ...state,
  userState: action.payload.movieUserState,
});

const setIsMovieFavoriteSuccess = (state: State, action) => {
  if (action.payload.id === state.details?.id) {
    return {
      ...state,
      userState: {
        ...state.userState,
        isFavorite: action.payload.isFavorite,
      },
    };
  }
  return state;
};

const setIsMovieInWatchlistSuccess = (state: State, action) => {
  if (action.payload.id === state.details?.id) {
    return {
      ...state,
      userState: {
        ...state.userState,
        isInWatchlist: action.payload.isInWatchlist,
      },
    };
  }
  return state;
};

const rateMovieSuccess = (state: State, action) => {
  if (action.payload.id === state.details?.id) {
    return {
      ...state,
      userState: {
        ...state.userState,
        rating: action.payload.rating,
      },
    };
  }
  return state;
};

const logoutSuccess = (state: State) => ({
  ...state,
  userState: initialState.userState,
});

const movieActionHandler = {
  [MovieActionTypes.FETCH_MOVIE_START]: fetchMovieStart,
  [MovieActionTypes.FETCH_MOVIE_SUCCESS]: fetchMovieSuccess,
  [MovieActionTypes.FETCH_MOVIE_USER_STATE_SUCCESS]: fetchMovieUserStateSuccess,
  [MovieActionTypes.SET_IS_MOVIE_FAVORITE_SUCCESS]: setIsMovieFavoriteSuccess,
  [MovieActionTypes.SET_IS_MOVIE_IN_WATCHLIST_SUCCESS]: setIsMovieInWatchlistSuccess,
  [MovieActionTypes.RATE_MOVIE_SUCCESS]: rateMovieSuccess,
  [AuthActionTypes.LOGOUT_SUCCESS]: logoutSuccess,
};

export default handleActions(movieActionHandler, initialState);
