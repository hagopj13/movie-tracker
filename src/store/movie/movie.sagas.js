import { all, call, takeLatest, put, select } from 'redux-saga/effects';

import * as api from 'api/tmdb';
import authSelectors from 'store/auth/auth.selectors';
import { convertResponseToMovie, convertResponseToMovieUserState } from 'api/tmdb/utils';

import MovieActionTypes from './movie.types';
import movieActions from './movie.actions';

export function* fetchMovie({ payload: { id } }) {
  yield put(movieActions.fetchMovieStart());
  const sessionId = yield select(authSelectors.selectSessionId);
  try {
    const { data } = yield call(api.getMovieDetails, id, sessionId);
    const movie = yield call(convertResponseToMovie, data);
    const movieUserState = yield call(convertResponseToMovieUserState, data.account_states);
    yield put(movieActions.fetchMovieSuccess(movie, movieUserState));
  } catch (error) {
    yield put(movieActions.fetchMovieFailure(error.status_message));
  }
}

export function* fetchMovieUserState({ payload: { id } }) {
  yield put(movieActions.fetchMovieUserStateStart());
  const sessionId = yield select(authSelectors.selectSessionId);
  try {
    const { data } = yield call(api.getMovieUserState, id, sessionId);
    const movieUserState = yield call(convertResponseToMovieUserState, data);
    yield put(movieActions.fetchMovieUserStateSuccess(movieUserState));
  } catch (error) {
    yield put(movieActions.fetchMovieUserStateFailure(error.status_message));
  }
}

export function* setIsMovieFavorite({ payload: { id, isFavorite } }) {
  yield put(movieActions.setIsMovieFavoriteStart());
  const sessionId = yield select(authSelectors.selectSessionId);
  const accountId = yield select(authSelectors.selectAccountId);
  try {
    yield call(api.setIsMovieFavorite, id, isFavorite, sessionId, accountId);
    yield put(movieActions.setIsMovieFavoriteSuccess(id, isFavorite));
  } catch (error) {
    yield put(movieActions.setIsMovieFavoriteFailure(error.status_message));
  }
}

export function* setIsMovieInWatchlist({ payload: { id, isInWatchlist } }) {
  yield put(movieActions.setIsMovieInWatchlistStart());
  const sessionId = yield select(authSelectors.selectSessionId);
  const accountId = yield select(authSelectors.selectAccountId);
  try {
    yield call(api.setIsMovieInWatchlist, id, isInWatchlist, sessionId, accountId);
    yield put(movieActions.setIsMovieInWatchlistSuccess(id, isInWatchlist));
  } catch (error) {
    yield put(movieActions.setIsMovieInWatchlistFailure(error.status_message));
  }
}

export function* rateMovie({ payload: { id, rating } }) {
  yield put(movieActions.rateMovieStart());
  const sessionId = yield select(authSelectors.selectSessionId);
  try {
    if (rating) {
      yield call(api.rateMovie, id, rating * 2, sessionId);
    } else {
      yield call(api.deleteMovieRating, id, sessionId);
    }
    yield put(movieActions.rateMovieSuccess(id, rating));
  } catch (error) {
    yield put(movieActions.rateMovieFailure(error.status_message));
  }
}

export function* onFetchMovie() {
  yield takeLatest(MovieActionTypes.FETCH_MOVIE, fetchMovie);
}

export function* onFetchMovieUserState() {
  yield takeLatest(MovieActionTypes.FETCH_MOVIE_USER_STATE, fetchMovieUserState);
}

export function* onSetIsMovieFavorite() {
  yield takeLatest(MovieActionTypes.SET_IS_MOVIE_FAVORITE, setIsMovieFavorite);
}

export function* onSetIsMovieInWatchlist() {
  yield takeLatest(MovieActionTypes.SET_IS_MOVIE_IN_WATCHLIST, setIsMovieInWatchlist);
}

export function* onRateMovie() {
  yield takeLatest(MovieActionTypes.RATE_MOVIE, rateMovie);
}

export default function* movieSagas() {
  yield all([
    call(onFetchMovie),
    call(onFetchMovieUserState),
    call(onSetIsMovieFavorite),
    call(onSetIsMovieInWatchlist),
    call(onRateMovie),
  ]);
}
