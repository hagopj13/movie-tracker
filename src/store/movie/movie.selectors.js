// @flow
import { createSelector } from 'reselect';

const selectMovie = (state) => state.movie;

const selectMovieDetails = createSelector([selectMovie], (movie) => movie.details);

const selectMovieUserState = createSelector([selectMovie], (movie) => movie.userState);

export default {
  selectMovieDetails,
  selectMovieUserState,
};
