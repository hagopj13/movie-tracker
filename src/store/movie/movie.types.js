export default {
  FETCH_MOVIE: 'FETCH_MOVIE',
  FETCH_MOVIE_START: 'FETCH_MOVIE_START',
  FETCH_MOVIE_SUCCESS: 'FETCH_MOVIE_SUCCESS',
  FETCH_MOVIE_FAILURE: 'FETCH_MOVIE_FAILURE',
  FETCH_MOVIE_USER_STATE: 'FETCH_MOVIE_USER_STATE',
  FETCH_MOVIE_USER_STATE_START: 'FETCH_MOVIE_USER_STATE_START',
  FETCH_MOVIE_USER_STATE_SUCCESS: 'FETCH_MOVIE_USER_STATE_SUCCESS',
  FETCH_MOVIE_USER_STATE_FAILURE: 'FETCH_MOVIE_USER_STATE_FAILURE',
  SET_IS_MOVIE_FAVORITE: 'SET_IS_MOVIE_FAVORITE',
  SET_IS_MOVIE_FAVORITE_START: 'SET_IS_MOVIE_FAVORITE_START',
  SET_IS_MOVIE_FAVORITE_SUCCESS: 'SET_IS_MOVIE_FAVORITE_SUCCESS',
  SET_IS_MOVIE_FAVORITE_FAILURE: 'SET_IS_MOVIE_FAVORITE_FAILURE',
  SET_IS_MOVIE_IN_WATCHLIST: 'SET_IS_MOVIE_IN_WATCHLIST',
  SET_IS_MOVIE_IN_WATCHLIST_START: 'SET_IS_MOVIE_IN_WATCHLIST_START',
  SET_IS_MOVIE_IN_WATCHLIST_SUCCESS: 'SET_IS_MOVIE_IN_WATCHLIST_SUCCESS',
  SET_IS_MOVIE_IN_WATCHLIST_FAILURE: 'SET_IS_MOVIE_IN_WATCHLIST_FAILURE',
  RATE_MOVIE: 'RATE_MOVIE',
  RATE_MOVIE_START: 'RATE_MOVIE_START',
  RATE_MOVIE_SUCCESS: 'RATE_MOVIE_SUCCESS',
  RATE_MOVIE_FAILURE: 'RATE_MOVIE_FAILURE',
};
