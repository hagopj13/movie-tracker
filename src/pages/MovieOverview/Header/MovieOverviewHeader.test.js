import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { details as movie } from 'store/fixtures/movie';

import MovieOverviewHeader from './MovieOverviewHeader';

describe('MovieOverviewHeader component', () => {
  let shallow;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    const mockProps = {
      movie,
    };
    wrapper = shallow(<MovieOverviewHeader {...mockProps} />);
  });

  it('should render the MovieOverviewHeader component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
