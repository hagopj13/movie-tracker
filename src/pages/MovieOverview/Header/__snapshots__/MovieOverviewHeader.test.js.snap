// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieOverviewHeader component should render the MovieOverviewHeader component correctly 1`] = `
<Connect(MovieBackdrop)
  backdropPath="/lWNASscWXn32Asr9vkB1wq6cKvD.jpg"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-poster-2"
      elevation={5}
    >
      <Connect(MoviePoster)
        height={450}
        posterPath="/y1MYZkwhZK6L0Jy4YMuPktzDOfn.jpg"
        width={300}
      />
    </WithStyles(ForwardRef(Card))>
    <div
      className="makeStyles-infoContainer-3"
    >
      <MovieSummary
        movie={
          Object {
            "actors": Array [
              Object {
                "character": "Sherlock Holmes",
                "id": 3223,
                "name": "Robert Downey Jr.",
                "profilePath": "/1YjdSym1jTG7xjHSI0yGGWEsw5i.jpg",
              },
              Object {
                "character": "Dr. John Watson",
                "id": 9642,
                "name": "Jude Law",
                "profilePath": "/aQD9e9SPjxCdyqxyXKBmEhNgjtR.jpg",
              },
              Object {
                "character": "Professor James Moriarty",
                "id": 15440,
                "name": "Jared Harris",
                "profilePath": "/mGODbYpkR1NKBOV3rtTfZ7EWoIl.jpg",
              },
            ],
            "backdropPath": "/lWNASscWXn32Asr9vkB1wq6cKvD.jpg",
            "genres": Array [
              Object {
                "id": 12,
                "name": "Adventure",
              },
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 80,
                "name": "Crime",
              },
              Object {
                "id": 9648,
                "name": "Mystery",
              },
            ],
            "id": 58574,
            "overview": "There is a new criminal mastermind at large (Professor Moriarty) and not only is he Holmes’ intellectual equal, but his capacity for evil and lack of conscience may give him an advantage over the  detective.",
            "posterPath": "/y1MYZkwhZK6L0Jy4YMuPktzDOfn.jpg",
            "recommendations": Array [
              Object {
                "backdropPath": "/kqvEVAvKZpw3MX3hNH6bVK6heBH.jpg",
                "id": 10528,
                "overview": "Eccentric consulting detective, Sherlock Holmes and Doctor John Watson battle to bring down a new nemesis and unravel a deadly plot that could destroy England.",
                "posterPath": "/momkKuWburNTqKBF6ez7rvhYVhE.jpg",
                "releaseDate": "2009-12-23",
                "title": "Sherlock Holmes",
                "voteAverage": 3.6,
                "voteCount": 10235,
              },
              Object {
                "backdropPath": "/bdNHNvZc9fM6Cw3r7HHMCWacI9I.jpg",
                "id": 10764,
                "overview": "Quantum of Solace continues the adventures of James Bond after Casino Royale. Betrayed by Vesper, the woman he loved, 007 fights the urge to make his latest mission personal. Pursuing his determination to uncover the truth, Bond and M interrogate Mr. White, who reveals that the organization that blackmailed Vesper is far more complex and dangerous than anyone had imagined.",
                "posterPath": "/f2rCskvSSr7ywPAnFGhk4FT6Piz.jpg",
                "releaseDate": "2008-10-30",
                "title": "Quantum of Solace",
                "voteAverage": 3.2,
                "voteCount": 5159,
              },
              Object {
                "backdropPath": "/yiggfsFRk5z2005oGojfyKkv7DI.jpg",
                "id": 20526,
                "overview": "Sam Flynn, the tech-savvy and daring son of Kevin Flynn, investigates his father's disappearance and is pulled into The Grid. With the help of a mysterious program named Quorra, Sam quests to stop evil dictator Clu from crossing into the real world.",
                "posterPath": "/xcO8Uun0mfmE3N9CaiCy0tvjpiu.jpg",
                "releaseDate": "2010-12-10",
                "title": "TRON: Legacy",
                "voteAverage": 3.2,
                "voteCount": 4944,
              },
            ],
            "releaseDate": "2011-11-22",
            "reviews": Array [
              Object {
                "author": "Author Name",
                "content": "As a general adventure and action movie I would consider this as a decent, even good, movie. As a Sherlock Holmes movie…not so much. Unfortunately, watching this movie confirms my suspicions. This is a Hollywood version of Sherlock Holmes. No style, no finesse. A Sherlock Holmes movie should not be a Hollywood action rollercoaster. It should be intelligent, stylish, witty perhaps involving detective work and, above all, British. It didn’t help that they enrolled Stephen Frey. This film is just not right.

I really cannot stand the hostility between Sherlock Holmes and Doctor Watson. There no justification for it in the works of Sir Arthur Conan Doyle. The slow motion scenes are nothing but annoying. The lame jokes are out of place in a Sherlock Holmes movie.

It gets a 5 out of 10 just because it’s actually a decent movie…if you can forget that they claim it to be a Sherlock Holmes one.",
                "id": "5af7018fc3a3681256009e81",
                "url": "https://www.themoviedb.org/review/5af7018fc3a3681256009e81",
              },
            ],
            "runtime": 129,
            "status": "Released",
            "tagline": "The game is afoot.",
            "title": "Sherlock Holmes: A Game of Shadows",
            "voteAverage": 3.6,
            "voteCount": 7326,
          }
        }
      />
      <div
        className="makeStyles-actionsContainer-4"
      >
        <Connect(MovieActions)
          movie={
            Object {
              "actors": Array [
                Object {
                  "character": "Sherlock Holmes",
                  "id": 3223,
                  "name": "Robert Downey Jr.",
                  "profilePath": "/1YjdSym1jTG7xjHSI0yGGWEsw5i.jpg",
                },
                Object {
                  "character": "Dr. John Watson",
                  "id": 9642,
                  "name": "Jude Law",
                  "profilePath": "/aQD9e9SPjxCdyqxyXKBmEhNgjtR.jpg",
                },
                Object {
                  "character": "Professor James Moriarty",
                  "id": 15440,
                  "name": "Jared Harris",
                  "profilePath": "/mGODbYpkR1NKBOV3rtTfZ7EWoIl.jpg",
                },
              ],
              "backdropPath": "/lWNASscWXn32Asr9vkB1wq6cKvD.jpg",
              "genres": Array [
                Object {
                  "id": 12,
                  "name": "Adventure",
                },
                Object {
                  "id": 28,
                  "name": "Action",
                },
                Object {
                  "id": 80,
                  "name": "Crime",
                },
                Object {
                  "id": 9648,
                  "name": "Mystery",
                },
              ],
              "id": 58574,
              "overview": "There is a new criminal mastermind at large (Professor Moriarty) and not only is he Holmes’ intellectual equal, but his capacity for evil and lack of conscience may give him an advantage over the  detective.",
              "posterPath": "/y1MYZkwhZK6L0Jy4YMuPktzDOfn.jpg",
              "recommendations": Array [
                Object {
                  "backdropPath": "/kqvEVAvKZpw3MX3hNH6bVK6heBH.jpg",
                  "id": 10528,
                  "overview": "Eccentric consulting detective, Sherlock Holmes and Doctor John Watson battle to bring down a new nemesis and unravel a deadly plot that could destroy England.",
                  "posterPath": "/momkKuWburNTqKBF6ez7rvhYVhE.jpg",
                  "releaseDate": "2009-12-23",
                  "title": "Sherlock Holmes",
                  "voteAverage": 3.6,
                  "voteCount": 10235,
                },
                Object {
                  "backdropPath": "/bdNHNvZc9fM6Cw3r7HHMCWacI9I.jpg",
                  "id": 10764,
                  "overview": "Quantum of Solace continues the adventures of James Bond after Casino Royale. Betrayed by Vesper, the woman he loved, 007 fights the urge to make his latest mission personal. Pursuing his determination to uncover the truth, Bond and M interrogate Mr. White, who reveals that the organization that blackmailed Vesper is far more complex and dangerous than anyone had imagined.",
                  "posterPath": "/f2rCskvSSr7ywPAnFGhk4FT6Piz.jpg",
                  "releaseDate": "2008-10-30",
                  "title": "Quantum of Solace",
                  "voteAverage": 3.2,
                  "voteCount": 5159,
                },
                Object {
                  "backdropPath": "/yiggfsFRk5z2005oGojfyKkv7DI.jpg",
                  "id": 20526,
                  "overview": "Sam Flynn, the tech-savvy and daring son of Kevin Flynn, investigates his father's disappearance and is pulled into The Grid. With the help of a mysterious program named Quorra, Sam quests to stop evil dictator Clu from crossing into the real world.",
                  "posterPath": "/xcO8Uun0mfmE3N9CaiCy0tvjpiu.jpg",
                  "releaseDate": "2010-12-10",
                  "title": "TRON: Legacy",
                  "voteAverage": 3.2,
                  "voteCount": 4944,
                },
              ],
              "releaseDate": "2011-11-22",
              "reviews": Array [
                Object {
                  "author": "Author Name",
                  "content": "As a general adventure and action movie I would consider this as a decent, even good, movie. As a Sherlock Holmes movie…not so much. Unfortunately, watching this movie confirms my suspicions. This is a Hollywood version of Sherlock Holmes. No style, no finesse. A Sherlock Holmes movie should not be a Hollywood action rollercoaster. It should be intelligent, stylish, witty perhaps involving detective work and, above all, British. It didn’t help that they enrolled Stephen Frey. This film is just not right.

I really cannot stand the hostility between Sherlock Holmes and Doctor Watson. There no justification for it in the works of Sir Arthur Conan Doyle. The slow motion scenes are nothing but annoying. The lame jokes are out of place in a Sherlock Holmes movie.

It gets a 5 out of 10 just because it’s actually a decent movie…if you can forget that they claim it to be a Sherlock Holmes one.",
                  "id": "5af7018fc3a3681256009e81",
                  "url": "https://www.themoviedb.org/review/5af7018fc3a3681256009e81",
                },
              ],
              "runtime": 129,
              "status": "Released",
              "tagline": "The game is afoot.",
              "title": "Sherlock Holmes: A Game of Shadows",
              "voteAverage": 3.6,
              "voteCount": 7326,
            }
          }
        />
      </div>
    </div>
  </WithStyles(ForwardRef(Container))>
</Connect(MovieBackdrop)>
`;
